### ISSUE TITLE
Elasticsearch built times increased by 12% over the past week  

### ISSUE THREAD

**Issue by atorok** [elastic/elasticsearch#41057]
Consider this intake build : 
   https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/2994/
   ran against e10f77934e9a750381c08f588195f397ad6d320a on the 7th and took 4h 13m  on a SLES 12 immutable CI worker. 

This is how the build times evolved ( looking at SLES only ) 

date| took| url
-- | -- | --
April 9th 2019, 23:38:43.382 | 305.66 | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/3046/console
April 8th 2019, 19:11:03.938 | 326.69 | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/3019/console
April 8th 2019, 18:04:11.532 | 362.11 | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/3017/console
April 8th 2019, 16:20:54.512 | 323.25 | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/3011/console
April 8th 2019, 13:12:21.287 | 318.58 | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/3003/console
April 7th 2019, 22:44:51.734 | 274.15 | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/2994/console
April 7th 2019, 12:05:05.025 | 263.42 | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/2990/console

Note that there are changes to the worker image that were expected to improve performance for 3046.

**Comment by iverase** [comment]
@hendrikmuhs you might be interested in this issue.

**Comment by atorok** [comment]
Here's the sample jobs and build times for CentOS 


April 10th 2019, 04:45:47.898 | 344.89 | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/3054/console
-- | -- | --

  | April 9th 2019, 21:28:46.165 | 340.88 | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/3043/console

  | April 9th 2019, 17:37:37.797 | 324.01 | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/3037/console

  | April 9th 2019, 16:03:52.234 | 335.65 | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/3036/console

  | April 8th 2019, 21:49:19.877 | 329.52 | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/3022/console

  | April 8th 2019, 21:37:43.808 | 329.08 | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/3021/console

  | April 8th 2019, 19:24:17.199 | 350.37 | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/3020/console

  | April 7th 2019, 00:25:38.841 | 291.46 | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/2988/console

  | April 5th 2019, 18:43:32.631 | 275.26 | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/2972/console

  | April 5th 2019, 15:28:06.225 | 304.67 | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/2969/console

  | April 5th 2019, 01:53:32.433 | 290.23 | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/2957/console

  | April 4th 2019, 22:17:38.099 | 275.46 | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/2956/console

**Comment by atorok** [comment]
Looking at the build scans for the previous  jobs  being compared, it seems like the tasks that take longest are around the same, but everything takes just a bit longer in the long build.

| task                                                                                           |   3054              |  2988
:x-pack:plugin:ml:qa:native-multi-node-tests:integTestRunner   |  12m 13.772s  | 9m 13.543s
:x-pack:plugin:ml:qa:ml-with-security:integTestRunner	            |  11m 50.843s  |  6m 57.956s  	 	                                                                            
:build-tools:integTest	                                                            |  8m 40.742s | 8m 18.708s	 	 
:x-pack:plugin:integTestRunner                                                   |	8m 32.027s | 6m 13.465s	  	 
:server:integTest	                                                                    | 8m 10.357s | 6m 34.539s

**Comment by atorok** [comment]
Running `./gradlew :x-pack:plugin:ml:qa:native-multi-node-tests:integTestRunner -Dtests.seed=884069A787BC42` on a warmed up Centos 7 CI worker image. 

```
BUILD SUCCESSFUL in 12m 11s (vm1)
BUILD SUCCESSFUL in 13m 12s (vm2)
BUILD SUCCESSFUL in 11m 58s (vm1)
BUILD SUCCESSFUL in 12m 18s (vm2)
```
There seems to be some variance on the time it takes for the tests to run even when trying to control all aspects.

Running the same against 51b9d5e from 2988 to account for any changes in the code. 
```
BUILD SUCCESSFUL in 14m 43s (vm1)
BUILD SUCCESSFUL in 14m 10s (vm2)
```
Strangely it looks to take even longer when the expectation would have been for it to be faster on this commit.

Running again with java 8 as runtime this time:
`./gradlew :x-pack:plugin:ml:qa:native-multi-node-tests:integTestRunner -Dtests.seed=884069A787BC42 -Djava.runtime=8`
```
BUILD SUCCESSFUL in 12m 3s (vm1)
BUILD SUCCESSFUL in 11m 19s (vm2)
```

In conclusion neither using an old commit that used to be fast, not using the old commit that used to be fast with runtime java 8 seems to make it as fast as when it ran in C, and there doesn't seem to be signs that things are slower.

**Comment by atorok** [comment]
Let's look at images as the next suspects running the same `./gradlew :x-pack:plugin:ml:qa:native-multi-node-tests:integTestRunner -Dtests.seed=884069A787BC42` command: 
 - April 3d: 
     -  BUILD SUCCESSFUL in 13m 30s  
     -  java 8: BUILD SUCCESSFUL in 13m 34s
 - April 6d: 
     - BUILD SUCCESSFUL in 13m 49s
     - java 8: 
 - April 7d:  
     -  BUILD SUCCESSFUL in 13m 50s
     - java 8: BUILD SUCCESSFUL in 13m 34s
 - April 8d: BUILD SUCCESSFUL in 14m 51s
     - BUILD SUCCESSFUL in 12m 43s
     - java 8: BUILD SUCCESSFUL in 13m 45s

From here, it doesn't seem to be significant differences in the images nor to they seem influenced by the java version used.

**Comment by atorok** [comment]
![Screenshot_2019-04-10_16-30-23](https://user-images.githubusercontent.com/2565652/55882693-a360a300-5bad-11e9-9fc6-a6449adb3452.png)

Looking a bit further back at how much builds took, it seems that  in fact the previous few days were  the anomaly in that the build ran faster, and then it came back up.
